### ISSUE TITLE
False positive: No name 'Iterable' in module 'collections'

### ISSUE THREAD

**Issue by jacebrowning** [PyCQA/pylint#2597]
### Steps to reproduce
Run `pylint` on a file containing:

```python
from collections import Iterable

print(isinstance(None, Iterable))
print(isinstance([42], Iterable))
print(isinstance("xy", Iterable))
```

### Current behavior

```
E0611: No name 'Iterable' in module 'collections' (no-name-in-module)
```

### Expected behavior

No errors should be displayed as `Iterable` is indeed importable from `collections`, though I'm not sure how as I don't see the class defined in the source code. Similar code is recommended on a popular Stack Overflow answer: https://stackoverflow.com/a/1952481/429533

### pylint --version output

```
pylint 2.1.1
astroid 2.1.0-dev
Python 3.7.0 (default, Jul  3 2018, 17:05:15)
[Clang 9.1.0 (clang-902.0.39.2)]
```

**Comment by PCManticore** [comment]
@jacebrowning I think you are getting this error because `pylint` doesn't understand that `collections.Iterable` is in fact located in `collections.abc.Iterable`. When running this script with python, there's also a warning:
```
a.py:1: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
```

Given that, I'm not sure if it's worth the effort to add support for find `collections.Iterable`.

**Comment by jacebrowning** [comment]
@PCManticore I didn't realize that `Iterable` had moved to `collections.abc`. Thanks for pointing that out! Changing the import to `from collections.abc import Iterable` works just fine.

Given that this problem completely goes away in 3.8, I agree that this isn't necessary to fix.
